From eaebec3db7f8469e8d22fd83a6b45dfff05fcea9 Mon Sep 17 00:00:00 2001
From: Greger Wrang <greger.wrang@westermo.se>
Date: Fri, 16 Nov 2018 08:18:04 +0100
Subject: [PATCH 32/38] route/link: Move LINK_ATTR_IFNAME to a proper location
Organization: Westermo Network Technologies AB

It should be set right where link name is set.

Signed-off-by: Greger Wrang <greger.wrang@westermo.se>
Signed-off-by: Joachim Nilsson <troglobit@gmail.com>
---
 lib/route/link.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/route/link.c b/lib/route/link.c
index 06e6d75..997f3e0 100644
--- a/lib/route/link.c
+++ b/lib/route/link.c
@@ -402,7 +402,7 @@ int rtnl_link_info_parse(struct rtnl_link *link, struct nlattr **tb)
 		return -NLE_MISSING_ATTR;
 
 	nla_strlcpy(link->l_name, tb[IFLA_IFNAME], IFNAMSIZ);
-
+	link->ce_mask |= LINK_ATTR_IFNAME;
 
 	if (tb[IFLA_STATS]) {
 		struct rtnl_link_stats *st = nla_data(tb[IFLA_STATS]);
@@ -624,7 +624,7 @@ static int __link_msg_parser(struct nlmsghdr *n, struct nl_parser_param *pp,
 	link->l_index = ifi->ifi_index;
 	link->l_flags = ifi->ifi_flags;
 	link->l_change = ifi->ifi_change;
-	link->ce_mask = (LINK_ATTR_IFNAME | LINK_ATTR_FAMILY |
+	link->ce_mask = (LINK_ATTR_FAMILY |
 			 LINK_ATTR_ARPTYPE| LINK_ATTR_IFINDEX |
 			 LINK_ATTR_FLAGS | LINK_ATTR_CHANGE);
 
-- 
2.25.1

